supervisor/core.py

```diff
@@ -305,6 +313,9 @@ async def shutdown(self):
         if self.state == CoreState.RUNNING:
             self.state = CoreState.SHUTDOWN
 
+        # Stop docker events monitor
+        await self.sys_docker_monitor.unload()
```

Comment:
Would be nice if we can handle that like hardware. Kinda `self.sys_docker.monitor` and then also only call `self.sys_docker.shutdown()` or `stop()` or whatever. Other than that, it looks great!

Reply:
So we can but the problem is `sys_docker` is a global. `DockerAPI` has no access to the `coresys` object by design and the monitor needs access to `coresys` in order to fire events on the bus. So I can have the monitor be an attribute of `DockerAPI` but `DockerAPI` cannot create it. Something else would have to create it and then add it to the `sys_docker` object.

That felt a little odd to me so I went with a separate `sys_docker_monitor`. But I'm open to changing to that approach, it's not a big change.

Reply:
You are right, DockerAPI doesn't have access to coresys, I guess history-wise. I would say, time to change that so we streamline the handling